types/src/multihash.rs

```diff
@@ -0,0 +1,214 @@
+use multihash_derive::{Hasher, MultihashDigest};
+use nmt_rs::simple_merkle::tree::MerkleHash;
+use nmt_rs::NamespaceMerkleHasher;
+
+use crate::byzantine::MULTIHASH_SHA256_NAMESPACE_FLAGGED_CODE;
+use crate::nmt::{
+    Namespace, NamespacedHashExt, NamespacedSha2Hasher, NAMESPACED_HASH_SIZE, NS_SIZE,
+};
+use crate::{Error, Result};
+
+pub type NamespacedHash = nmt_rs::NamespacedHash<NS_SIZE>;
+
+struct Sha256NamespaceFlaggedHasher {
+    hasher: NamespacedSha2Hasher,
+    hash: Option<NamespacedHash>,
+    hash_buf: [u8; NAMESPACED_HASH_SIZE],
+}
+
+impl Sha256NamespaceFlaggedHasher {
+    fn hash_data(&mut self, data: &[u8]) -> Result<()> {
+        let hash = if data.len() == NAMESPACED_HASH_SIZE * 2 {
+            let (left, right) = data.split_at(NAMESPACED_HASH_SIZE);
+            self.hasher
+                .hash_nodes(&left.try_into()?, &right.try_into()?)
```

Comment:
after the len check we can just unwrap there

Reply:
actually after we unwrap there, this method doesn't need to return the result

Reply:
True - switched to `unwrap` and removed the `Result`.